react/features/local-recording/recording/flac/FlacAdapter.js

```diff
@@ -0,0 +1,170 @@
+                    logger.debug('libflac is ready.');
+                    resolve();
+                } else {
+                    logger.error(
+                        `Unknown event
+                        from encoder (WebWorker): "${e.data.command}"!`);
+                }
+            };
+
+            this._encoder.postMessage({
+                command: MAIN_THREAD_INIT,
+                config: {
+                    sampleRate: 44100,
+                    bps: 16
+                }
+            });
+        });
+
+        const callbackInitAudioContext = (resolve, reject) => {
+            navigator.getUserMedia(
+
+                // constraints - only audio needed for this app
+                {
+                    audioBitsPerSecond: 44100, // 44 kbps
+                    audio: true,
+                    mimeType: 'application/ogg' // useless?
+                },
+
+                // Success callback
+                stream => {
+                    this._audioContext = new AudioContext();
+                    this._audioSource
+                     = this._audioContext.createMediaStreamSource(stream);
+                    this._audioProcessingNode
+                      = this._audioContext.createScriptProcessor(4096, 1, 1);
+                    this._audioProcessingNode.onaudioprocess = e => {
+                        // delegate to the WebWorker to do the encoding
+                        const channelLeft = e.inputBuffer.getChannelData(0);
+
+                        this._encoder.postMessage({
+                            command: MAIN_THREAD_NEW_DATA_ARRIVED,
+                            buf: channelLeft
+                        });
+                    };
+                    logger.debug('AudioContext is set up.');
+                    resolve();
+                },
+
+                // Error callback
+                err => {
+                    logger.error(`Error calling getUserMedia(): ${err}`);
+                    reject();
+                }
+            );
+        };
+
+        // FIXME: because Promise constructor immediately executes the executor
+        // function. This is undesirable, we want callbackInitAudioContext to be
+        // executed only **after** promiseInitWorker is resolved.
+        return promiseInitWorker
+            .then(() => new Promise(callbackInitAudioContext));
+    }
+
+    /**
+     * Implements {@link RecordingAdapter#start()}.
+     *
+     * @inheritdoc
+     */
+    start() {
+        this._audioSource.connect(this._audioProcessingNode);
+        this._audioProcessingNode.connect(this._audioContext.destination);
+    }
+
+    /**
+     * Implements {@link RecordingAdapter#stop()}.
+     *
+     * @inheritdoc
+     */
+    stop() {
+        return new Promise(resolve => {
+            this._audioProcessingNode.onaudioprocess = undefined;
+            this._audioProcessingNode.disconnect();
+            this._audioSource.disconnect();
```

Comment:
This doesn't matter but I'm pointing it out: if stop is called before initialization then this will error. I just wanted to point that out for no real purpose other than that I noticed.